### PR TITLE
Merge csv main logic

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5667

## What this PR does / why we need it:
handle metric/log data compaction, merge multi files in one, reduce qps cost in read or scan.
1. add config `MergeCycle` ctrl Merge as service.
2. init `Merge` as service in `main.go` while `MergeCycle > 0`
3. add `export/merge.go` handle merge main logic
    - `NewMergeService` init merge as service, with param `serviceInit` to avoid multi init.
    - `NewMerge` handle merge obj init.
    - `Merge::Start` as service loop, trigger `Merge::Main` each cycle
    - `Merge::Main` handle handle job,
        1. foreach account, build `rootPath` with tuple {account, date, Table }
        2. call `Merge::doMergeFiles` with all files in `rootPath`,  do merge job
    - `Merge::doMergeFiles` handle one job flow: read each file, merge in cache, write into file. 
4. move `trace/table.go` to `export/table.go`, as common util for pkg `util/metic` and `util/trace` to maintain schema.

TODO: run merge as cron task

### CODE REVIEW

File: pkg/util/export/merge.go [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+	"github.com/matrixorigin/simdcsv"
+)
+
+// ========================
+// handle merge
+// ========================
+
+// Merge like a compaction, merge input files into one/two/... files.
+type Merge struct {
+	Table       *Table                  // see WithTable
```
Comment (fengttt) [inline pkg/util/export/merge.go:39]: Well, yeah, I went to see WithTable, then nothing there :)

File: pkg/util/export/type.go [hunk 2]
```diff
@@ -34,6 +34,15 @@ func init() {
 	SetDefaultContextFunc(func() context.Context { return context.Background() })
 }
 
+const (
```
Comment (fengttt) [inline pkg/util/export/type.go:37]: I have seen this in many places.   While it is fine, maybe we should use the one in Common.

**Review by fengttt — COMMENTED** [review]
I am starting to lose memory of our discussions.

So write to CSV, will write one csv file, or a few csv files every few sends.   Then when we merge we want to split them to different tables.   Is this still the plan?

File: pkg/util/export/type.go (hunk 2 above)
Comment (xzxiong) [reply pkg/util/export/type.go:37]: got it.
I will `pkg/common/constant/byte.go` handle const value `Byte, KB, MB, GB, ...`.
and try to handle other pkgs in new pr.

Comment (xzxiong) [reply pkg/util/export/type.go:37]: i find `KB, MB, GB, TB, ...` already defined in `pkg/common/mpool/mpool.go`
should i just use it, or mv it to `pkg/common/constant/byte.go` ?

Comment (fengttt) [reply pkg/util/export/type.go:37]: You can move it, your call. :)

**Review by fengttt — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by aptend — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

**Comment by xzxiong** [comment]
@mergifyio refresh

**Comment by xzxiong** [comment]
@mergifyio refresh

**Pull request merged**